tests/migrations/test_operations.py

```diff
@@ -1801,6 +1801,244 @@ def inner_method(models, schema_editor):
+            "Article",
+            [
+                ("id", models.AutoField(primary_key=True)),
+                ("blog", models.ForeignKey(to="test_blog.Blog", on_delete=models.CASCADE)),
+                ("name", models.CharField(max_length=100)),
+                ("data", models.TextField(default="")),
+            ],
+            options={},
+        )
+        fill_initial_data = migrations.RunPython(initial_data, initial_data)
+        fill_big_data = migrations.RunPython(big_data, big_data)
+
+        grow_article_id = migrations.AlterField(
+            "Article",
+            "id",
+            models.BigAutoField(primary_key=True)
+        )
+
+        grow_blog_id = migrations.AlterField(
+            "Blog",
+            "id",
+            models.BigAutoField(primary_key=True)
+        )
+
+        project_state = ProjectState()
+        new_state = project_state.clone()
+        with connection.schema_editor() as editor:
+            create_blog.state_forwards("test_blog", new_state)
+            create_blog.database_forwards("test_blog", editor, project_state, new_state)
+
+        project_state = new_state
+        new_state = new_state.clone()
+        with connection.schema_editor() as editor:
+            create_article.state_forwards("test_article", new_state)
+            create_article.database_forwards("test_article", editor, project_state, new_state)
+
+        project_state = new_state
+        new_state = new_state.clone()
+        with connection.schema_editor() as editor:
+            fill_initial_data.state_forwards("fill_initial_data", new_state)
+            fill_initial_data.database_forwards("fill_initial_data", editor, project_state, new_state)
+
+        project_state = new_state
+        new_state = new_state.clone()
+        with connection.schema_editor() as editor:
+            grow_article_id.state_forwards("test_article", new_state)
+            grow_article_id.database_forwards("test_article", editor, project_state, new_state)
+
+        state = new_state.clone()
+        article = state.apps.get_model("test_article.Article")
+        self.assertTrue(isinstance(article._meta.pk, models.BigAutoField))
+
+        project_state = new_state
+        new_state = new_state.clone()
+        with connection.schema_editor() as editor:
+            grow_blog_id.state_forwards("test_blog", new_state)
+            grow_blog_id.database_forwards("test_blog", editor, project_state, new_state)
+
+        state = new_state.clone()
+        blog = state.apps.get_model("test_blog.Blog")
+        self.assertTrue(isinstance(blog._meta.pk, models.BigAutoField))
+
+        project_state = new_state
+        new_state = new_state.clone()
+        with connection.schema_editor() as editor:
+            fill_big_data.state_forwards("fill_big_data", new_state)
+            fill_big_data.database_forwards("fill_big_data", editor, project_state, new_state)
+
+    def test_m2m_intermediate_table_is_not_migrated_properly(self):
```

Comment:
Yes, Django has an unfixed bug: many to many "middle" table is not migrated at all.